src/runtime_src/core/pcie/tools/xbflash.qspi/pcidev.cpp

```diff
@@ -97,6 +101,38 @@ pci_device(const std::string& sysfs, int ubar, size_t flash_off, std::string fla
   dev = d;
   func = f;
   user_bar_size = 0;
+
+  std::snprintf(sysfsname, sizeof(sysfsname), "%04x:%02x:%02x.%x",
+    domain, bus, dev, func);
+  std::string conffile("/sys/bus/pci/devices/");
+  conffile += sysfsname;
+  conffile += "/config";
+  
+  int conf_handle = ::open(conffile.c_str(), O_RDWR | O_SYNC);
+  if (conf_handle < 0) {
+    int err = errno;
+    std::cout << "Failed to open " << conffile << " : "
+     << strerror(err) << std::endl;
```

Comment:
if there is error, throw runtime_error so that the program doesn't not go forward